bench/workgen/CMakeLists.txt

```diff
@@ -0,0 +1,156 @@
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG 3.0 COMPONENTS python REQUIRED)
+include(${SWIG_USE_FILE})
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+
+# Determine which wiredtiger target we should use to link against the
+# Python Workgen API. For linking purposes, we need to ensure the wiredtiger
+# library was compiled as position independent code (fPIC). This being achieved
+# either using the 'WITH_PIC' configuration option or building a shared library
+# version of WiredTiger ('ENABLE_SHARED'). Default to whichever option is available
+# (with a preference for a PIC static build), otherwise throw an error if neither
+# build is available.
+set(wiredtiger_target)
+if(ENABLE_STATIC AND WITH_PIC)
+    set(wiredtiger_target wiredtiger_static)
+elseif(ENABLE_SHARED)
+    set(wiredtiger_target wiredtiger_shared)
+else()
+    message(STATIC "Skipping workgen compiliation: Requires either a shared library (ENABLE_SHARED)"
+                    "or a PIC-enabled (WITH_PIC) build of wiredtiger")
+endif()
+
+set(python_libs)
+set(python_version)
+set(python_executable)
+# Source the users Python3 install, extracting the Python 3 version, library and interpreter install paths.
+source_python3_package(python_libs python_version python_executable)
+
+# Compile the intermediate C++ workgen sources into static library.
+add_library(
+    workgen_cxx
+    STATIC
+        ${CMAKE_CURRENT_SOURCE_DIR}/workgen.cxx
+        ${CMAKE_CURRENT_SOURCE_DIR}/workgen_func.c
+)
+target_include_directories(
+    workgen_cxx
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_BINARY_DIR}/include
+        ${CMAKE_BINARY_DIR}/config
+        ${CMAKE_SOURCE_DIR}/src/include
+        ${CMAKE_SOURCE_DIR}/test/utility
+)
+set_property(TARGET workgen_cxx PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+# Manually specify the interface name '_workgen' over a generated name.
+# Our workgen tooling expects the swig interface name '_workgen'
+# when importing the generated swig library.
+list(APPEND swig_flags "-c++")
+list(APPEND swig_flags "-interface;_workgen")
+list(APPEND swig_flags "-threads")
+list(APPEND swig_flags "-O")
+list(APPEND swig_flags "-Wall")
+# Pass in our generated include directories.
+list(APPEND swig_flags "-I${CMAKE_BINARY_DIR}/include")
+list(APPEND swig_flags "-I${CMAKE_BINARY_DIR}/config")
+
+set(CMAKE_SWIG_FLAGS ${swig_flags})
+# By default, the CMake SWIG module expects the swig interface file extension to be ".i". Newer
+# versions of CMake 3.14+ support specifying 'SWIG_SOURCE_FILE_EXTENSIONS' to consider different
+# extensions. Though given our minimum support version is 3.10, we need to copy the file (and its header dependency)
+# over to our binary directory using the ".i" extension.
+configure_file(${CMAKE_CURRENT_LIST_DIR}/workgen.h ${CMAKE_CURRENT_BINARY_DIR}/workgen.h COPYONLY)
+configure_file(${CMAKE_CURRENT_LIST_DIR}/workgen.swig ${CMAKE_CURRENT_BINARY_DIR}/workgen.i COPYONLY)
```

Comment:
If it's more convenient or standard, you could rename `workgen.swig` to be `workgen.i` .  I used `.swig` to distinguish from C include files that were named with `.i`, but include files have been renamed to be `_inline.h`.  Though `.swig` may be more readable for anyone casually browsing directories.

Reply:
Thanks for background and clarification! Using  `workgen.i` would make it much more convenient for CMake (both to meet its expectations that SWIG interface files conventionally use the `.i` extension and to track the original source  as a file-level dependency). I've updated the PR, renaming the file to `workgen.i`.

Reply:
@alisonfel, just checked your change for `.swig` -> `.i`, looks good.